src/shared/udev-util.h

```diff
@@ -46,3 +51,19 @@ int udev_resolve_subsys_kernel(const char *string, char *result, size_t maxsize,
 
 int udev_queue_is_empty(void);
 int udev_queue_init(void);
+
+#if HAVE_SYS_SDT_H
+
+#define DEVICE_TRACE_POINT(name, dev, ...)                                     \
+        do {                                                                   \
+                PROTECT_ERRNO;                                                 \
+                const char *_s = NULL;                                        \
+                sd_device *_d = dev;                                           \
+                sd_device_action_t _a = _SD_DEVICE_ACTION_INVALID;             \
+                (void) sd_device_get_action(_d, &_a);                          \
+                (void) sd_device_get_sysname(_d, &_s);                         \
```

Comment:
hmm, i figure we also want to include the sysfs path and the subsystem here.

Reply:
Ok, I've added those